frontend/src/main/scala/bloop/exec/JvmProcessForker.scala

```diff
@@ -113,23 +117,65 @@ final class JvmForker(config: JdkConfig, classpath: Array[AbsolutePath]) extends
       extraClasspath: Array[AbsolutePath]
   ): Task[Int] = {
     val jvmOptions = jargs.map(_.stripPrefix("-J")) ++ config.javaOptions
-    val fullClasspath = (classpath ++ extraClasspath).map(_.syntax).mkString(pathSeparator)
-    Task.fromTry(javaExecutable).flatMap { java =>
-      val classpathOption = "-cp" :: fullClasspath :: Nil
-      val appOptions = mainClass :: args.toList
-      val cmd = java.syntax :: jvmOptions.toList ::: classpathOption ::: appOptions
-      val logTask = if (logger.isVerbose) {
+    val fullClasspath = classpath ++ extraClasspath
+
+    for {
+      fullClasspathStr <- Task.fromTry(ensureClasspathLength(fullClasspath))
+      java <- Task.fromTry(javaExecutable)
+
+      classpathOption = "-cp" :: fullClasspathStr :: Nil
+      appOptions = mainClass :: args.toList
+      cmd = java.syntax :: jvmOptions.toList ::: classpathOption ::: appOptions
+
+      logTask <- if (logger.isVerbose) {
         val debugOptions =
           s"""
              |Fork options:
              |   command      = '${cmd.mkString(" ")}'
              |   cwd          = '$cwd'""".stripMargin
         Task(logger.debug(debugOptions)(DebugFilter.All))
-      } else Task.unit
-      logTask.flatMap(_ => Forker.run(cwd, cmd, logger, opts))
+      } else {
+        Task.unit
+      }
+      res <- Forker.run(cwd, cmd, logger, opts)
+    } yield {
+      res
     }
   }
 
+  private def ensureClasspathLength(classpath: Array[AbsolutePath]): Try[String] = {
+    val fullClasspathStr = classpath.map(_.syntax).mkString(pathSeparator)
+
+    if (fullClasspathStr.length > JvmProcessForker.classpathCharLimit)
```

Comment:
Should we include the length of JVM options and program arguments instead of only `fullClasspathStr.length`? This check may still not insure us against "argument list too large" if the program argument list is very large.

Reply:
we might want to write another test with a very large argument list

Reply:
Is there a way to stuff arguments into a MANIFEST jar, because the answer to that seems to be "no".

Reply:
Might have to use your solution with a wrapper class!

Reply:
Descoping to just classpath shortening